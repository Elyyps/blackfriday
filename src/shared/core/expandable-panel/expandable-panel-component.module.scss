ul,
p {
  margin: 0 auto;
}
.expandable-panel {
  cursor: pointer;

  &__list {
    margin: 0px;
    display: contents;
    cursor: pointer;
    text-align: left;
    margin: 0px;
    list-style: none;
  }

  &__block {
    position: relative;
    z-index: 1;
  }
}
.expandable-panel__block {
  .expandable-panel__content {
    overflow: hidden;
    max-height: 0;
    transition: max-height 0.25s ease-in-out, margin 0.3s ease-in-out;
    margin: 0;
    z-index: 1;
    position: relative;
    a {
      font-family: $global-font-family;
      font-size: 14px;
    }
    cursor: initial;
    ul:not([class]) > li:not(:first-child) {
      margin-top: 16px;
    }
  }
  &__open {
    .expandable-panel__content {
      margin: 24px 0 0 0;
      max-height: 500px;
      height: auto;
      transition: max-height 0.25s ease-in-out, margin 0.3s ease-in-out;
    }
  }
}
