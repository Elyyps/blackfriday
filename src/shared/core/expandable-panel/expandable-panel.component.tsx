import * as React from "react";
import style from "./expandable-panel-component.module.scss";
import classNames from "classnames";
import { IconComponent } from "../icon";

import ArrowUp from "@assets/icons/chevron-up.svg";
import ArrowDown from "@assets/icons/chevron-down.svg";

interface IExpandablePanelComponentProps {
  callback?: (identifier?: string) => void;
  children?: JSX.Element;
  identifier?: string;
  mainContent: JSX.Element;
  opened?: boolean;
}

const ExpandablePanelComponent = (props: IExpandablePanelComponentProps) => {
  const invokeCallback = () => {
    if (props.callback) {
      props.callback(props.identifier);
    }
  };

  return (
    <React.Fragment>
      <div className={style["expandable-panel"]}>
        <div
          role="listitem"
          className={classNames(
            `${style["expandable-panel__block"]}`,
            props.opened && `${style["expandable-panel__block__open"]}`
          )}
        >
          <ul className={style["expandable-panel__list"]}>
            <li>
              <div
                role="button"
                aria-label="expandButton"
                onClick={invokeCallback}
                className={`${style["expandable-panel__title"]}`}
              >
                <div className={style["expandable-panel__icon"]}>
                  <IconComponent icon={props.opened ? ArrowUp : ArrowDown} size="20px" fillColor={"#fff"} />
                </div>

                {props.mainContent}
              </div>
              <div
                className={classNames(
                  `${style["expandable-panel__content"]}`,
                  !props.opened && `${style["expandable-panel__hidden"]}`
                )}
              >
                {props.children}
              </div>
            </li>
          </ul>
        </div>
      </div>
    </React.Fragment>
  );
};

export { ExpandablePanelComponent };
