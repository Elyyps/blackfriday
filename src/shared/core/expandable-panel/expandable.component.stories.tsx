import * as React from "react";
import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { ExpandablePanelComponent } from "./expandable-panel.component";

const notes = require("./expandable.md");

storiesOf("Expandable", module)
  .addDecorator(withA11y)
  .add("Basic implementation", () => <ExpandablePanelComponent mainContent={<span>Title</span>} />, {
    notes
  });
