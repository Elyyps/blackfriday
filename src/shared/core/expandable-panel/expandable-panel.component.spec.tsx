import React from "react";
import { shallow } from "enzyme";
import { ExpandablePanelComponent } from "./expandable-panel.component";

describe("[ExpandablePanelComponent]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <ExpandablePanelComponent mainContent={<span>Title</span>} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
