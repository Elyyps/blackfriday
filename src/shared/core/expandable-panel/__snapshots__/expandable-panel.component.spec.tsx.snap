// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ExpandablePanelComponent] should render component without crashing 1`] = `
<Fragment>
  <div
    className="expandable-panel"
  >
    <div
      className="expandable-panel__block"
      role="list"
    >
      <ul
        className="expandable-panel__list"
      >
        <li
          aria-label="expanded panel"
          onClick={[Function]}
          role="listitem"
        >
          <span>
            Title
          </span>
          <div
            className="expandable-panel__content expandable-panel__hidden"
          />
        </li>
      </ul>
    </div>
  </div>
</Fragment>
`;
